scripts/src/java/org/oppia/android/scripts/maven/GenerateMavenDependenciesList.kt

```diff
@@ -0,0 +1,413 @@
+  val finalUpdatedList = mutableListOf<MavenDependency>()
+
+  latestDependenciesList.forEach { mavenDependency ->
+    val updateLicenseList = mutableListOf<License>()
+    var numberOfLicensesThatRequireHumanEffort = 0
+    val origin = mavenDependency.originOfLicense
+    mavenDependency.licenseList.forEach { license ->
+      val updatedLicense = finalLicensesSet.find { it.primaryLink == license.primaryLink }
+      if (updatedLicense != null) {
+        updateLicenseList.add(updatedLicense)
+        if (
+          updatedLicense.primaryLinkType == PrimaryLinkType.NEEDS_INTERVENTION ||
+          updatedLicense.primaryLinkType == PrimaryLinkType.SCRAPE_FROM_LOCAL_COPY
+        ) {
+          numberOfLicensesThatRequireHumanEffort++
+        }
+      } else {
+        if (license.primaryLinkType == PrimaryLinkType.NEEDS_INTERVENTION) {
+          numberOfLicensesThatRequireHumanEffort++
+        }
+        updateLicenseList.add(license)
+      }
+    }
+    val dependency = MavenDependency
+      .newBuilder()
+      .setArtifactName(mavenDependency.artifactName)
+      .setArtifactVersion(mavenDependency.artifactVersion)
+      .addAllLicense(updateLicenseList)
+    if (origin != OriginOfLicenses.UNKNOWN) {
+      dependency.setOriginOfLicense(origin)
+    } else {
+      if (updateLicenseList.isNotEmpty()) {
+        if (numberOfLicensesThatRequireHumanEffort == updateLicenseList.size) {
+          dependency.originOfLicense = OriginOfLicenses.MANUAL
+        } else if (numberOfLicensesThatRequireHumanEffort == 0) {
+          dependency.originOfLicense = OriginOfLicenses.ENTIRELY_FROM_POM
+        } else if (numberOfLicensesThatRequireHumanEffort != updateLicenseList.size) {
+          dependency.originOfLicense = OriginOfLicenses.PARTIALLY_FROM_POM
+        }
+      }
+    }
+    finalUpdatedList.add(dependency.build())
+  }
+  return finalUpdatedList
+}
+
+/** Retrieves the list of [MavenDependency] from maven_dependencies.textproto. */
+private fun retrieveMavenDependencyList(): List<MavenDependency> {
+  return getProto(
+    "maven_dependencies.pb",
+    MavenDependencyList.getDefaultInstance()
+  ).mavenDependencyList.toList()
+}
+
+/**
+ * Helper function to parse the textproto file to a proto class.
+ *
+ * @param textProtoFileName name of the textproto file to be parsed
+ * @param proto instance of the proto class
+ * @return proto class from the parsed textproto file
+ */
+private fun getProto(
+  textProtoFileName: String,
+  proto: MavenDependencyList
+): MavenDependencyList {
+  val protoBinaryFile = File("scripts/assets/$textProtoFileName")
+  val builder = proto.newBuilderForType()
+  val protoObject = FileInputStream(protoBinaryFile).use {
+    builder.mergeFrom(it)
+  }.build() as MavenDependencyList
+  return protoObject
+}
+
+private fun parseArtifactName(artifactName: String): String {
+  var colonIndex = artifactName.length - 1
+  while (artifactName.isNotEmpty() && artifactName[colonIndex] != ':') {
```

Comment:
Is this actually correct? I thought we needed to count the number of colons in the coord to determine if there even is a version present since not all coordinates have versions specified.

Reply:
Actually, I didn't find any example where the version is not mentioned in the coordinate. And if that's the case, then how the count of colons can help us determine if there is a version of the dependency or not.
Can there be a maximum of one colon in the coordinate?

Other than that I also have another thought here, can we avoid mentioning the version at this moment as we are already keeping the full name of the dependency. 
We can extract the version of the dependency from the dependency name itself (if there exists one), this will save us writing a complete list of versions in the XML. WDYT?

Reply:
I think the important requirement to meet is that if the version of a dependency changes, we may need to reevaluate the licenses list to make sure it's up-to-date.

Per https://maven.apache.org/pom.html#Maven_Coordinates I think we can just assume that there are always 2 colons and that coordinates are fully well-formed (all situations should result in an exception being thrown).

Reply:
Done.

Reply:
I just tried this and caught the exception for the artifact - `android.arch.core:common:jar:sources:1.1.1`- 
```
/home/prayutsu/.cache/bazel/_bazel_prayutsu/65c630ff6a7321024a3fd83ac1880b0b/execroot/__main__/bazel-out/k8-fastbuild/bin/scripts/generate_maven_dependencies_list: line 112: fg: no job control
Exception in thread "main" java.lang.Exception: Couldn't parse the version for the artifact 'android.arch.core:common:jar:sources:1.1.1'
        at org.oppia.android.scripts.maven.MavenDependenciesListWriter$Companion.omitVersionAndReplaceColonsHyphensPeriods(MavenDependenciesListWriter.kt:256)
        at org.oppia.android.scripts.maven.MavenDependenciesListWriter$Companion.getDependencyListFromMavenInstall(MavenDependenciesListWriter.kt:336)
        at org.oppia.android.scripts.maven.MavenDependenciesListWriter$Companion.main(MavenDependenciesListWriter.kt:62)
        at org.oppia.android.scripts.maven.GenerateMavenDependenciesListKt.main(GenerateMavenDependenciesList.kt:8)

```

Reply:
Discussed during the team chat a method for doing this.